src/update/UpdateManager.cpp

```diff
@@ -0,0 +1,92 @@
+#include "UpdateManager.h"
+#include "UpdatePopupDialog.h"
+
+#include "widgets/AudacityMessageBox.h"
+
+#include <wx/platinfo.h>
+#include <wx/utils.h>
+
+static const char* prefsUpdatePopupDialogShown = "/Update/UpdatePopupDialogShown";
+
+enum { ID_TIMER = wxID_HIGHEST + 1 };
+
+BEGIN_EVENT_TABLE(UpdateManager, wxEvtHandler)
+    EVT_TIMER(ID_TIMER, UpdateManager::OnTimer)
+END_EVENT_TABLE()
+
+UpdateManager::UpdateManager(AudacityProject& project)
+{
+    wxFrame* parrentFrame = project.GetFrame();
+    wxASSERT(parrentFrame);
+    
+    mParent = reinterpret_cast<wxWindow*> (parrentFrame);
+    
+    getUpdates();
+    
+    mTimer.SetOwner(this, ID_TIMER);
+    mTimer.StartOnce(kTimerInterval);
+}
+
+UpdateManager::~UpdateManager()
+{}
+
+void UpdateManager::enableNotification(bool enable)
+{
+    gPrefs->Write(prefsUpdatePopupDialogShown, enable);
+}
+
+bool UpdateManager::isNotificationEnabled()
+{
+    return gPrefs->ReadBool(prefsUpdatePopupDialogShown, false);
+}
+
+VersionPatch UpdateManager::getVersionPatch() const
+{
+    return mVersionPatch;
+}
+
+void UpdateManager::getUpdates()
+{
+    ServerCommunication::UpdateDataFormat updateResponse;
+    
+    ServerCommunication communicator;
+    if (!communicator.getUpdateData(&updateResponse))
+    {
+        // TODO: remake text and may be Parrent.
+        AudacityMessageBox(
+            XO("Unable to connect on Audacity update server."),
+            XO("Error update check"),
+            wxOK | wxCENTRE,
+            mParent);
```

Comment:
You use mParent just so that a modal dialog has some parent window.  But a modal dialog is allowed to have a null parent.

The dialog has a stack-like lifetime and is destroyed before that AudacityMessageBox call is completed.

(A modeless dialog can also have a null parent, but you must then do extra work to be sure it is destroyed at some time.)

I see that this code is rewritten in later commits, but the same remarks apply:

`
            mParent->CallAfter([this] {
                UpdatePopupDialog dlg(mParent, this);
                const int code = dlg.ShowModal();
`

So I say:  don't remember the mParent pointer.  Instead, find a window just when you need it to show the modal dialog.  And if you can't, just use null.

Use FindProjectFrame( GetActiveProject() ).

Include <wx/frame.h>, so the compiler knows wxFrame is-a wxWindow, rather than using a pointer cast.